device.c

```diff
@@ -186,7 +187,19 @@ static int vcam_try_fmt_vid_cap(struct file *file,
         }
         dev->output_format.sizeimage =
             dev->output_format.bytesperline * dev->output_format.height;
+
+        /* resize the videobuffer */
         vcam_update_format_cap(dev, false);
+
+        /* resize the framebuffer */
+        destroy_framebuffer((const char *) dev->vcam_fb_fname);
```

Comment:
`destroy_framebuffer` is considered harmful. It might be associated with different proc entries after new `init_framebuffer` requst.

Reply:
Check this project: https://github.com/hadi77ir/linux-module-virtfb
You can use `fbset` to change its mode/resolution on the fly.